modules/cloudevent-recorder/recorder.tf

```diff
@@ -13,14 +13,15 @@ resource "google_service_account" "recorder" {
 // The recorder service account is the only identity that should be writing
 // to the regional GCS buckets.
 resource "google_storage_bucket_iam_binding" "recorder-writes-to-gcs-buckets" {
-  for_each = var.regions
+  for_each = var.method == "trigger" ? var.regions : {}
 
   bucket  = google_storage_bucket.recorder[each.key].name
   role    = "roles/storage.admin"
   members = ["serviceAccount:${google_service_account.recorder.email}"]
 }
 
 module "this" {
+  count      = var.method == "trigger" ? 1 : 0
```

Comment:
This is a breaking change anyways that will trigger a recreation of the resources, so why not just go for the migration?